cmd/tnf/addclaim/addclaim.go

```diff
@@ -27,12 +29,230 @@ var (
+	log.Info("claim1 and claim2 has diff RawResults ", slist)
+	log.Info("test name that claim1 has but claim 2 dont has", r)
+	return nil
+}
+
+func unmarshalClaimFile(calimdata []byte) (Cni, HwInfo, RawResult, error) {
+	var cni Cni
+	var hwinfo HwInfo
+	var rawResult RawResult
+
+	errcni := json.Unmarshal(calimdata, &cni)
+	if errcni != nil {
+		log.Fatalf("Error in unmarshal the cni from claim2 file  :%v", errcni)
+		return cni, hwinfo, rawResult, errcni
+	}
+	// csi
+	var csi Csi
+	errcsi := json.Unmarshal(calimdata, &csi)
+	if errcsi != nil {
+		log.Fatalf("Error in unmarshal the csi from cliam1 file  :%v", errcsi)
+		return cni, hwinfo, rawResult, errcsi
+	}
+
+	// HwInfo
+	errhwinfo := json.Unmarshal(calimdata, &hwinfo)
+	if errhwinfo != nil {
+		log.Fatalf("Error in unmarshal the hwinfo from cliam1 file  :%v", errhwinfo)
+		return cni, hwinfo, rawResult, errhwinfo
+	}
+
+	// rawResult
+	errrawResult := json.Unmarshal(calimdata, &rawResult)
+	if errrawResult != nil {
+		log.Fatalf("Error in unmarshal the rawResult from cliam1 file  :%v", errrawResult)
+		return cni, hwinfo, rawResult, errrawResult
+	}
+	return cni, hwinfo, rawResult, nil
+}
+
+func compare2Hwinfo(hwinfo1, hwinfo2 map[string]interface{}) {
+	var nodes1, nodes2 []string
+
+	for key := range hwinfo1 {
+		nodes1 = append(nodes1, key)
+	}
+	for key := range hwinfo2 {
+		nodes2 = append(nodes2, key)
+	}
+	fmt.Println("nodes2 and nodes diffs", missing(nodes2, nodes1))
+}
+
+func compare2TestCaseResults(testcaseResult1, testcaseResult2 testcase) (diffresult testcase, notFoundtest []string) {
+	for _, result1 := range testcaseResult1 {
+		findeName := false
+		for _, result2 := range testcaseResult2 {
+			if result2.Name == result1.Name {
+				findeName = true
+				if (result2.Status) != (result1.Status) {
+					diffresult = append(diffresult, result1)
+				}
+				break
+			}
+		}
+		if !findeName {
+			notFoundtest = append(notFoundtest, result1.Name)
+		}
+	}
+	return diffresult, notFoundtest
+}
+
+// empty struct (0 bytes)
+type void struct{}
+
+// missing compares two slices and returns slice of differences
+func missing(a, b []string) []string {
+	// create map with length of the 'a' slice
+	ma := make(map[string]void, len(a))
+	diffs := []string{}
+	// Convert first slice to map with empty struct (0 bytes)
+	for _, ka := range a {
+		ma[ka] = void{}
+	}
+	// find missing values in a
+	for _, kb := range b {
+		if _, ok := ma[kb]; !ok {
+			diffs = append(diffs, kb)
+		}
+	}
+	return diffs
+}
+
+func compare2cni(cni1, cni2 map[string]cnistruct) {
+	for node, val := range cni1 {
+		for node2, val2 := range cni2 {
+			if node == node2 {
+				c, s := compare2cniHelper(val, val2)
+				if len(s) != 0 {
+					log.Info("node ", node2, " cnis found in claim1 but not present in claim2: ", s)
+				}
+				if len(c) != 0 {
+					log.Info("node ", node2, " cnis present in both claim 1 and 2 but with different plugins: ", c)
+				}
+			}
```

Comment:
You should add a `break` at the end of this `if node == node2`, as you already found the element.